addons/sys_rack/fnc_rackPFH.sqf

```diff
@@ -0,0 +1,73 @@
+/*
+ * Author: ACRE2Team
+ * Per frame execution. Sets if player is inside a vehicle and manages the access to rack radios.
+ *
+ * Arguments:
+ * 0: Array of arguments <ARRAY>
+ *  0: Player unit <OBJECT>
+ *  1: Vehicle with intercom <OBJECT>
+ * 1: PFH unique identifier <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [[player, vehicle player], 12] call acre_sys_intercom_fnc_intercomPFH
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_param", "_handle"];
+
+_param params ["_player", "_vehicle"];
+
+// Check if the player entered a position with a rack already active in intercom
+if (_player != vehicle _player) then {
+    {
+        private _radioId = [_x] call FUNC(getMountedRadio);
+        if (_radioId != "" && {!(_radioId in ACRE_ACCESSIBLE_RACK_RADIOS || _radioId in ACRE_HEARABLE_RACK_RADIOS)}) then {
+            private _functionality = [_radioId, _vehicle, _player, _x] call EFUNC(sys_intercom,getRxTxCapabilities);
+            if (_functionality != RACK_NO_MONITOR) then {
+                // Add the radio to the active list since it is already active in the intercom system
+                [_vehicle, _player, _radioId] call FUNC(startUsingMountedRadio);
+            };
+        };
+    } forEach (([_vehicle, _player] call FUNC(getHearableVehicleRacks)) apply {toLower _x});
+};
+
+// Check whether the vehicle rack radios can still be used
+private _remove = [];
+{
+    if (!([_x] call EFUNC(sys_radio,radioExists))) then {_remove pushBackUnique _x;};
+    private _rack = [_x] call FUNC(getRackFromRadio);
+    if (_rack == "") then { _remove pushBackUnique _x; }; // Radio is no longer stored in a rack
+
+    private _isRackHearable = [_rack, _player] call FUNC(isRackHearable);
+    private _isRackAccessible = [_rack, _player] call FUNC(isRackAccessible);
+
+    // Check only those radios connected on intercom systems
+    if (count ([_rack] call FUNC(getWiredIntercoms)) > 0 && _isRackHearable) then {
+        private _functionality = [_x, _vehicle, _player, toLower _rack] call EFUNC(sys_intercom,getRxTxCapabilities);
+
+        if (_functionality == RACK_NO_MONITOR) then {_remove pushBackUnique _x;};
+    };
+
+    if (!(_isRackAccessible || _isRackHearable)) then {_remove pushBackUnique _x; };
+} forEach (ACRE_ACCESSIBLE_RACK_RADIOS + ACRE_HEARABLE_RACK_RADIOS);
+
+{
+    if (_x in ACRE_ACCESSIBLE_RACK_RADIOS) then {
+        ACRE_ACCESSIBLE_RACK_RADIOS = ACRE_ACCESSIBLE_RACK_RADIOS - [_x];
```

Comment:
Would probably get nice improvement by using `deleteAt + find`, especially since this can be ran quiet often.

Reply:
```
x1 = [1, 2, 3];
x1 = x1 - [2];
-> 0.0014 ms
```
```
x1 = [1, 2, 3];
x1 deleteAt (x1 find 2);
-> 0.0012 to 0.0014 ms
```
```
x1 = [2];
x1 = x1 - [2];
-> 0.0012 to 0.0014 ms
```
```
x1 = [2];
x1 deleteAt (x1 find 2);
-> 0.0011 to 0.0012 ms
```

Reply:
this after 10k iterations. For the scope in here we are then looking at: no speed improvement or between 0.1 to 0.2 microseconds. Which transforms into none to 0.02 nanoseconds when executing this code once a second.

Reply:
Yes, but considering that it is not worse, we should still use it, it's still an enhancement. Even if we don't use many racks at once here, someone might, whatever mod.